lib/bibtex/names.y

```diff
@@ -134,7 +157,8 @@ require 'strscan'
   def scan(input)
     @src = StringScanner.new(input)
     @brace_level = 0
-    @stack = []
+    @last_and = 0
+    @stack = [[:ANDFL,'#dummy']]
```

Comment:
In the commit f5084da, I used `(^begin)`. I tried some erroneous cases, and the error messages made sense.